### ISSUE TITLE
False positive S1121 in with expressions

### ISSUE THREAD

**Issue by YairHalberstadt** [SonarSource/sonar-dotnet#4047]
### Description

S1121 warns when a variable is assigned inside an expression.

C# 9.0 introduced `with` expressions. This always triggers this rule, even though it's not an assignment - it's creating a new object with the specified changes.

### Repro steps

```csharp
var a = new A(1) with { B = 2 };

record A(int B){}
```

### Expected behavior

no warning

### Actual behavior

warning

**Comment by pavel-mikula-sonarsource** [comment]
Hi @YairHalberstadt,

Thank you for reporting this case, I can confirm it as a False Positive. We already have this case documented in your unit tests and we'll add support for C# 9 Records in MMF-2223 so I'll close this issue for now.
